blues/models.py

```diff
@@ -0,0 +1,317 @@
+
+        self.n_dartboard = n_dartboard
+        self.dartboard = dartboard
+
+    def calc_from_center(self, com):
+
+        distList = []
+        diffList = []
+        indexList = []
+        #Find the distances of the COM to each dart, appending
+        #the results to distList
+        for dart in self.dartboard:
+            diff = com - dart
+            dist = np.sqrt(np.sum((diff)*(diff)))*unit.nanometers
+            distList.append(dist)
+            diffList.append(diff)
+        selected = []
+        #Find the dart(s) less than self.dart_size
+        for index, entry in enumerate(distList):
+            if entry <= self.dart_size:
+                selected.append(entry)
+                diff = diffList[index]
+                indexList.append(index)
+        #Dart error checking
+        #to ensure reversibility the COM should only be
+        #within self.dart_size of one dart
+        if len(selected) == 1:
+            return selected[0], diffList[indexList[0]]
+        elif len(selected) == 0:
+            return None, diff
+        elif len(selected) >= 2:
+            #COM should never be within two different darts
+            raise ValueError('sphere size overlap, check darts')
+            #TODO can treat cases using appropriate probablility correction
+
+    def n_findDart(self, nc_context):
+        """
+        Helper function to dynamically update dart positions based on positions
+        of other particles.
+        Arguments
+        ---------
+        basis_particles: list of 3 ints
+            Specifies the 3 indices of particles whose coordinates will be used
+            as basis vectors. If None is specified, uses those found in basis particles
+        Returns
+        -------
+        dart_list list of 1x3 np.arrays in units.nm
+            new dart positions calculated from the particle_pairs
+            and particle_weights
+        """
+        basis_particles = self.basis_particles
+        #make sure there's an equal number of particle pair lists
+        #and particle weight lists
+        dart_list = []
+        state_info = nc_context.getState(True, True, False, True, True, False)
+        temp_pos = state_info.getPositions(asNumpy=True)
+        part1 = temp_pos[basis_particles[0]]
+        part2 = temp_pos[basis_particles[1]]
+        part3 = temp_pos[basis_particles[2]]
+        for dart in self.n_dartboard:
+            old_center = findOldCoord(part1, part2, part3, dart)
+            dart_list.append(old_center)
+        self.dartboard = dart_list[:]
+        return dart_list
+
+    def reDart(self, changevec):
+        """
+        Helper function to choose a random dart and determine the vector
+        that would translate the COM to that dart center
+        """
+        dartindex = np.random.randint(len(self.dartboard))
+        dvector = self.dartboard[dartindex]
+        chboard = dvector + changevec
+        return chboard
+
+    def smartDartMove(self, nc_context):
```

Comment:
Much like how the `random_rotation` function is under the `ncmc.MoveProposal` class, I think we should move this function under the same place to keep things consistent. The Model class will store the properties of the ligand (like positions/indices) while the MoveProposal class handles functionality of the movement.